frontend-react/src/components/USLink.tsx

```diff
@@ -1,38 +1,103 @@
-import React, { PropsWithChildren } from "react";
+import React, { AnchorHTMLAttributes } from "react";
 import { Link, NavLink } from "react-router-dom";
+import classnames from "classnames";
 
-// Known issue with the `PropsWithChildren generic in React 18,
-// so I wrote this in a way that we can just remove `<any>` and be
-// okay if we update our React version.
-interface USNavLinkProps extends PropsWithChildren<any> {
-    href: string;
+/** React.PropsWithChildren has known issues with generic extension in React 18,
+ * so rather than using it here, we are using our own definition of child types.
+ * One less headache when updating to React 18 in the future! */
+interface CustomLinkProps {
+    children: React.ReactNode;
+    anchor?: boolean;
     className?: string;
     activeClassName?: string;
 }
-type USLinkProps = Omit<USNavLinkProps, "activeClassName">;
+type USLinkProps = AnchorHTMLAttributes<{}> &
+    Omit<CustomLinkProps, "activeClassName">;
+type USNavLinkProps = Pick<AnchorHTMLAttributes<{}>, "href"> & CustomLinkProps;
 
-/** A single link for rendering standard links */
-export const USLink = ({ children, href, className }: USLinkProps) => (
-    <Link to={href} className={`usa-link ${className}`}>
-        {children}
-    </Link>
-);
+/** A single link for rendering standard links. Uses a `Link` by default
+ * but adding `anchor` will make this a generic anchor tag.
+ * @example
+ * <USLink href="/page">To Page</USLink> // uses <Link> from react-router-dom
+ * <USLink anchor href="#this-section-on-my-page">To Section</USLink> // uses <a>
+ * */
+export const USLink = ({
+    anchor = false,
+    children,
+    className,
+    href,
```

Comment:
VERY minor point, but I'm a little wary of calling this prop `href` because it's not technically a 1:1 with the underlying `href` HTML attribute for Links.  If we were to add `basename`s to our routing, something like `href="/whatever"` could ultimately turn into `href="/base/name/whatever"` when rendered.

Reply:
Nah, this was a pedantic point about clarity of naming -- I was just wary of making it seem like the `href` prop would be guaranteed to be 1:1 with the rendered `href` data attribute, which isn't necessarily correct in the router context:

```jsx
<Router basename="/blah">
    <USLink href="/whatever">Whatever</USLink>
</Router>
```

The anchor that's actually rendered is `<a href="/blah/whatever">Whatever</a>` as opposed to `<a href="/whatever">Whatever</a>` so the `href` prop could be a little misleading since it's not exactly the resultant `href` attribute value.  Definitely not a blocking point, though!

Reply:
Ahhh, got it!